src/common/Utility.js

```diff
@@ -0,0 +1,65 @@
+"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define([], factory);
+    } else {
+        root.common_Utility = factory();
+    }
+}(this, function () {
+
+    /**
+     *
+     */
+
+    function _naturalSort(a, b, order, idx, sortCaseSensitive) {
+        var re = /(^([+\-]?(?:0|[1-9]\d*)(?:\.\d*)?(?:[eE][+\-]?\d+)?)?$|^0x[0-9a-f]+$|\d+)/gi,
+            sre = /(^[ ]*|[ ]*$)/g,
+            dre = /(^([\w ]+,?[\w ]+)?[\w ]+,?[\w ]+\d+:\d+(:\d+)?[\w ]?|^\d{1,4}[\/\-]\d{1,4}[\/\-]\d{1,4}|^\w+, \w+ \d+, \d{4})/,
+            hre = /^0x[0-9a-f]+$/i,
+            ore = /^0/,
+            i = function(s) { return !sortCaseSensitive && ('' + s).toLowerCase() || '' + s; },
+            // convert all to strings strip whitespace
+            x = i(idx ? a[idx] : a).replace(sre, '') || '',
+            y = i(idx ? b[idx] : b).replace(sre, '') || '',
+            // chunk/tokenize
+            xN = x.replace(re, '\0$1\0').replace(/\0$/,'').replace(/^\0/,'').split('\0'),
+            yN = y.replace(re, '\0$1\0').replace(/\0$/,'').replace(/^\0/,'').split('\0'),
+            // numeric, hex or date detection
+            xD = parseInt(x.match(hre), 16) || (xN.length !== 1 && x.match(dre) && Date.parse(x)),
+            yD = parseInt(y.match(hre), 16) || xD && y.match(dre) && Date.parse(y) || null,
+            oFxNcL, oFyNcL;
+        // first try and sort Hex codes or Dates
+        if (yD) {
+            if ( xD < yD ) { return order === "ascending" ? -1 : 1; }
+            else if ( xD > yD ) { return order === "ascending" ? 1 : -1; }
+        }
+        // natural sorting through split numeric strings and default strings
+        for(var cLoc=0, numS=Math.max(xN.length, yN.length); cLoc < numS; cLoc++) {
+            // find floats not starting with '0', string or 0 if not defined (Clint Priest)
+            oFxNcL = !(xN[cLoc] || '').match(ore) && parseFloat(xN[cLoc]) || xN[cLoc] || 0;
+            oFyNcL = !(yN[cLoc] || '').match(ore) && parseFloat(yN[cLoc]) || yN[cLoc] || 0;
+            // handle numeric vs string comparison - number < string - (Kyle Adams)
+            if (isNaN(oFxNcL) !== isNaN(oFyNcL)) { return (isNaN(oFxNcL)) ? 1 : -1; }
+            // rely on string comparison if different types - i.e. '02' < 2 != '02' < '2'
+            else if (typeof oFxNcL !== typeof oFyNcL) {
+                oFxNcL += '';
+                oFyNcL += '';
+            }
+            if (oFxNcL < oFyNcL) { return order === "ascending" ? -1 : 1; }
+            if (oFxNcL > oFyNcL) { return order === "ascending" ? 1 : -1; }
+        }
+        return 0;
+    }
+
+    return {
+        /*
+         * USAGE: this._data = Utility.sort(this._data, context._currentSortOrder === 1 ? "ascending" : "descending", idx, true);
```

Comment:
I am not sure how I feel about this comment - it does have some useful information (ascending and descending are two options for "order") but the user still needs to dig into the code to workout how to use the function (which is fine IMO).